### ISSUE TITLE
[PRE REVIEW]: RayTracer.jl: A General Purpose Differentiable Renderer

### ISSUE THREAD

**Issue by whedon** [JuliaCon/proceedings-review#31]
**Submitting author:** @avik-pal (<a href="http://orcid.org/0000-0002-3938-7375">Avik Pal</a>)
**Repository:** <a href="https://github.com/avik-pal/RayTracer.jl" target ="_blank">https://github.com/avik-pal/RayTracer.jl</a>
**Editor:** @matbesancon
**Reviewers:** @snoeyink, @dourouc05

**Author instructions**

Thanks for submitting your paper to JuliaCon @avik-pal. The JuliaCon editor @matbesancon, will work with you on this issue to find a reviewer for your submission before creating the main review issue.

@avik-pal if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](http://bit.ly/2V9GSFq) have already agreed to review for JuliaCon and may be suitable for this submission.

**Editor instructions**

The JuliaCon submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @matbesancon it looks like you're currently assigned as the editor for this paper :tada:

For a list of things I can do to help you, just type:

```
@whedon commands
```

**What happens now?**

This submission is currently in a `pre-review` state which means we are waiting for the assigned editor (@matbesancon) to find some reviewers for your submission. This may take anything between a few hours to a couple of weeks. Thanks for your patience :smile_cat:

You can help the editor by looking at [this list of potential reviewers](http://bit.ly/2V9GSFq) to identify individuals who might be able to review your submission (please start at the bottom of the list). Also, feel free to suggest individuals who are not on this list by mentioning their GitHub handles here.

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #31 with the following error: 

 Can't find any papers to compile :-(

**Comment by matbesancon** [comment]
Fixing #27

**Comment by matbesancon** [comment]
@whedon add  @snoeyink as reviewer

**Comment by whedon** [comment]
OK,  @snoeyink is now a reviewer

**Comment by matbesancon** [comment]
@whedon generate pdf from branch juliacon-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch juliacon-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00031/jcon.00031/10.21105.jcon.00031.pdf)

**Comment by snoeyink** [comment]
This paper introduces a raytracer built in Julia that allows the Zygote Automatic Differentiation tool to compute gradients of parameters.
One can then optimize parameter values for scene understanding or reconstruction.

The examples shown are toy experiments of obtaining camera or light parameters to match a rendered image. They demonstrate that
the combination works, which is good.

I would like to see some discussion of more interesting parameters, such as determining material properties from images. And more on
limitations: determining geometry seems out of reach because there are so many discrete choices: the problem is not differentiable.
How fast is the gradient computation after AD? (How fast is the AD?) How do these grow with scene complexity?

This paper shows the hallmarks of being written in haste as the package is being developed.  I have many comments and suggestions in pdf uploaded to the package repository.

The abstract discusses training times for ML, which is not in the paper, so here is a rewrite:
In this paper we present RayTracer.jl, a renderer in Julia that is fully differentiable via Zygote.jl. This means that RayTracer not only renders 2D images from 3D scene parameters, but it can be used to optimize for model parameters that generate a target image in a Differentiable Programming (DP) pipeline. We interface our renderer with the deep learning library Flux for use in combination with neural networks. We demonstrate the use of this differentiable renderer in rendering tasks and in solving inverse graphics problems.

**Comment by snoeyink** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

```

**Comment by avik-pal** [comment]
@whedon generate pdf from branch juliacon-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch juliacon-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00031/jcon.00031/10.21105.jcon.00031.pdf)

**Comment by matbesancon** [comment]
@dehann @dourouc05 would you be available to review this proceeding?

**Comment by dourouc05** [comment]
I would be available, starting from Wednesday! (For the process, shall I put the review here? Or is there a more detailed guide?)

**Comment by matbesancon** [comment]
@whedon add @dourouc05 as reviewer

**Comment by whedon** [comment]
OK, @dourouc05 is now a reviewer

**Comment by matbesancon** [comment]
> For the process, shall I put the review here? Or is there a more detailed guide?

Yes there is a guide [here](https://proceedings.juliacon.org/guide/reviewers/), the review issue will be opened soon with a checklist

**Comment by matbesancon** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/JuliaCon/proceedings-review/issues/37. Feel free to close this issue now!
